frontend/src/metabase/nav/components/SearchBar.unit.spec.tsx

```diff
@@ -0,0 +1,182 @@
+import { Route } from "react-router";
+import userEvent from "@testing-library/user-event";
+import { waitFor, renderWithProviders, screen, within } from "__support__/ui";
+import {
+  setupRecentViewsEndpoints,
+  setupSearchEndpoints,
+} from "__support__/server-mocks";
+import {
+  createMockCollectionItem,
+  createMockModelObject,
+  createMockRecentItem,
+} from "metabase-types/api/mocks";
+import {
+  createMockSettingsState,
+  createMockState,
+} from "metabase-types/store/mocks";
+import { CollectionItem, RecentItem } from "metabase-types/api";
+import SearchBar from "metabase/nav/components/SearchBar";
+
+const TEST_SEARCH_RESULTS: CollectionItem[] = [
+  "Card ABC",
+  "Card BCD",
+  "Card DEF",
+  "Card EFG",
+].map((name, index) =>
+  createMockCollectionItem({
+    name,
+    id: index + 1,
+    getUrl: () => "/",
+  }),
+);
+
+const TEST_RECENT_VIEWS_RESULTS: RecentItem[] = [
+  "Recents ABC",
+  "Recents BCD",
+  "Recents CDE",
+  "Recents DEF",
+].map((name, index) =>
+  createMockRecentItem({
+    model_object: createMockModelObject({ name }),
+    model_id: index,
+  }),
+);
+
+const setup = ({
+  initialRoute = "/",
+  searchResultItems = TEST_SEARCH_RESULTS,
+  recentViewsItems = TEST_RECENT_VIEWS_RESULTS,
+} = {}) => {
+  const state = createMockState({
+    settings: createMockSettingsState({
+      "search-typeahead-enabled": true,
+    }),
+  });
+
+  setupSearchEndpoints(searchResultItems);
+  setupRecentViewsEndpoints(recentViewsItems);
+
+  const { history } = renderWithProviders(
+    <Route path="*" component={SearchBar} />,
+    {
+      withRouter: true,
+      initialRoute,
+      storeInitialState: state,
+    },
+  );
+
+  return { history };
+};
+
+const getSearchBar = () => {
+  return screen.getByPlaceholderText("Search…");
+};
+
+describe("SearchBar", () => {
+  describe("typing a search query", () => {
+    it("should change URL when user types a query and hits `Enter`", async () => {
+      const { history } = setup();
+
+      userEvent.type(getSearchBar(), "BC{enter}");
+
+      const location = history?.getCurrentLocation();
+      expect(location?.pathname).toEqual("search");
+      expect(location?.search).toEqual("?q=BC");
+    });
+    it("should render 'No Results Found' when the query has no results", async () => {
+      setup({ searchResultItems: [] });
+      const searchBar = getSearchBar();
+      userEvent.click(searchBar);
+      userEvent.type(searchBar, "XXXXX");
+      await waitFor(() =>
+        expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument(),
+      );
+
+      expect(
+        within(screen.getByTestId("search-bar-results-container")).getByText(
```

Comment:
Not sure `within` is needed. I think when running a unit/integration test like this. We only render relatively small UI components. `within` definitely is needed for E2E, but maybe not for unit tests? What do you think?

Obviously, it will be necessary if the component under test contains more than 1 instance of this text.